gdal/frmts/tiledb/tiledbdataset.cpp

```diff
@@ -0,0 +1,682 @@
+        tiledb::Stats::enable();
+
+    auto status = query.submit();
+
+    if ( poGDS->bStats )
+    {
+        tiledb::Stats::dump(stdout);
+        tiledb::Stats::disable();
+    }
+
+    query.finalize();
+
+    if (status != tiledb::Query::Status::COMPLETE)
+        return CE_Failure;
+    else
+        return CE_None;
+}
+
+/************************************************************************/
+/*                       GetColorInterpretation()                       */
+/************************************************************************/
+
+GDALColorInterp TileDBRasterBand::GetColorInterpretation()
+
+{
+    if ( nBand == 1 )
+        return GCI_RedBand;
+
+    else if( nBand == 2 )
+        return GCI_GreenBand;
+
+    else if ( nBand == 3 )
+        return GCI_BlueBand;
+
+    return GCI_AlphaBand;
+}
+
+/************************************************************************/
+/* ==================================================================== */
+/*                             TileDBDataset                            */
+/* ==================================================================== */
+/************************************************************************/
+
+/************************************************************************/
+/*                           ~TileDBDataset()                           */
+/************************************************************************/
+
+TileDBDataset::~TileDBDataset()
+
+{
+    FlushCache();
+    m_array->close();
+}
+
+/************************************************************************/
+/*                           AddFilter()                                */
+/************************************************************************/
+
+CPLErr TileDBDataset::AddFilter( const char* pszFilterName, const int level )
+
+{
+   if (pszFilterName == nullptr)
+        m_filterList->add_filter( tiledb::Filter( *m_ctx, TILEDB_FILTER_NONE )
+            .set_option( TILEDB_COMPRESSION_LEVEL, level ) );
+    else if EQUAL(pszFilterName, "GZIP")
+        m_filterList->add_filter( tiledb::Filter( *m_ctx, TILEDB_FILTER_GZIP )
+            .set_option( TILEDB_COMPRESSION_LEVEL, level ) );
+    else if EQUAL(pszFilterName, "ZSTD")
+        m_filterList->add_filter( tiledb::Filter( *m_ctx, TILEDB_FILTER_ZSTD )
+            .set_option( TILEDB_COMPRESSION_LEVEL, level ) );
+    else if EQUAL(pszFilterName, "LZ4")
+        m_filterList->add_filter( tiledb::Filter( *m_ctx, TILEDB_FILTER_LZ4 )
+            .set_option( TILEDB_COMPRESSION_LEVEL, level ) );
+    else if EQUAL(pszFilterName, "RLE")
+        m_filterList->add_filter( tiledb::Filter( *m_ctx, TILEDB_FILTER_RLE )
+            .set_option( TILEDB_COMPRESSION_LEVEL, level ) );
+    else if EQUAL(pszFilterName, "BZIP2")
+        m_filterList->add_filter( tiledb::Filter( *m_ctx, TILEDB_FILTER_BZIP2 )
+            .set_option( TILEDB_COMPRESSION_LEVEL, level ) );
+    else if EQUAL(pszFilterName, "DOUBLE-DELTA")
```

Comment:
Is there a reason to not include `POSITIVE-DELTA` also?

Reply:
added `POSITIVE-DELTA` to options